src/App/docutils/ComponentPreview/index.js

```diff
@@ -121,11 +121,29 @@ const ComponentPreview = ({ children, language, removeOuterTag, hideValue, remov
                 return "update switchcase!";
         }
 
+        const copyToClipboard = e => {
+            const code = e.target.parentElement.parentElement.querySelector(".code-view-code pre").innerText;
+
+            navigator.clipboard.writeText(code);
+
+            e.target.dataset.tooltip = "Copied!";
+        };
+
+        const tooltipContent = e => {
+            const code = e.target.parentElement.parentElement.querySelector(".code-view-code pre").innerText;
+
+            navigator.clipboard.readText().then(copied => {
+                e.target.dataset.tooltip = copied === code ? "Copied!" : "Copy to clipboard";
+            });
+
+        };
```

Comment:
It works now, which is great, but what if we change the DOM? Hard coding selectors based on event usually gives some kind of maintenance down the road.
Would it be possible to pass the code that we are going to copy directly into the functions instead of the event?
`onClick={copyToClipboard(code)}`